pages/instructorFileEditor/instructorFileEditor.ejs

```diff
@@ -1,78 +1,155 @@
-                        <button type="button" class="close" data-dismiss="alert" aria-label="Close">
-                            <span aria-hidden="true">&times;</span>
-                        </button>
+            <div class="card">
+                <div class="card-header">
+                    <div class="row align-items-center justify-content-between">
+                        <div class="col-auto">
+                            Editing file <%= fileEdit.dirName %>/<%= fileEdit.fileName %>
+                        </div>
+                        <div class="col-auto">
+                            <button id="file-editor-<%= fileEdit.uuid %>-save-button" name="__action" value="save_and_sync" class="btn btn-primary" disabled>Save and Sync</button>
+                        </div>
                     </div>
-                <% } %>
 
-                <% if (fileEdit.didFindOutdated) { %>
-                    <div class="alert alert-warning alert-dismissible fade show" role="alert">
-                        The file you are trying to edit was changed by another user since your last saved draft. You must click "Revert to Original" to continue editing. Doing so will discard your saved draft. If you have made any changes that you would like to preserve, you should copy the text of your saved draft before proceeding.
-                        <button type="button" class="close" data-dismiss="alert" aria-label="Close">
-                            <span aria-hidden="true">&times;</span>
-                        </button>
+                </div>
+                <div class="card-body p-0 row">
+                    <div class="container">
+                        <% if (fileEdit.alertResults) { %>
+                        <div class="alert <% if (fileEdit.didSave && fileEdit.didSync) { %>alert-success<% } else { %>alert-danger<% } %> alert-dismissible fade show m-2" role="alert">
+                            <div class="row align-items-center">
+                                <div class="col-auto">
+                                    <% if (fileEdit.didSave) { %>
+                                        <% if (fileEdit.didSync) { %>
+                                            File was both saved and synced successfully.
+                                        <% } else { %>
+                                            File was saved, but failed to sync.
+                                        <% } %>
+                                    <% } else { %>
+                                        Failed to save and sync file.
+                                    <% } %>
+                                </div>
+                                <% if (fileEdit.jobSequenceId != null) { %>
+                                <div class="col-auto">
+                                    <a class="btn btn-secondary btn-sm" data-toggle="collapse" href="#results-<%= fileEdit.uuid %>" role="button">Detail</a>
```

Comment:
Maybe change “Detail” to “Show details of error”?

Reply:
Maybe an extra-small button for these “detail” buttons would look better?

Reply:
I chose to show the results both on success and failure - that's why I'm not referring to "error" in the button (adding a switch b/w "Show details of success" and "Show details of error" or whatever seems unnecessary). Totally open to a better choice than "Detail".

With respect to extra-small buttons - that's no longer a default option in BS4 (as far as I understand, anyway).

Reply:
We do have quite a few `btn-xs` classes in our templates, and they seem to work. This is just a very weak suggestion, though.

Fair enough on “error”. I’d prefer “Show details”. Is it possible to add a close-icon button inside the collapsable that will close it? The alternative is to label the details-button with “Show/hide details”, but I’ve always found that a bit ugly. Of course we could add some JS to change “Show” to “Hide”, but I think that’s more mess than is warranted.

Way down in the weeds here, of course, so just do whatever you want (or nothing)!

Reply:
`btn-xs` is defined in a custom class, so I'll not use it unless there's a compelling reason.

The show/hide thing is done, in a small bit of JS.